es-app/src/views/gamelist/GridGameListView.cpp

```diff
@@ -20,27 +88,64 @@ FileData* GridGameListView::getCursor()
 
 void GridGameListView::setCursor(FileData* file)
 {
-	if(!mGrid.setCursor(file))
+	if(!mGrid.setCursor(file) && mGrid.getEntryCount() > 0)
 	{
-		populateList(file->getParent()->getChildrenListToDisplay());
+		populateList(file->getParent()->getChildren());
 		mGrid.setCursor(file);
+		mTitle.setText(file->getName());
 	}
 }
 
 bool GridGameListView::input(InputConfig* config, Input input)
 {
-	if(config->isMappedTo("left", input) || config->isMappedTo("right", input))
+	if (config->isMappedTo("left", input) || config->isMappedTo("right", input)
+	    || config->isMappedTo("up", input) || config->isMappedTo("down", input)) {
+		// Quick and dirty way of changing header title without doing in update()
+		mTitle.setText(mGrid.getSelectedName());
+		updateInfoPanel();
+
+		// Destroy dpad input so mGrid can use it.
 		return GuiComponent::input(config, input);
+	}
 
 	return ISimpleGameListView::input(config, input);
 }
 
+void GridGameListView::update(int deltatime) {
+	if (!bFocused)
+		return;
+
+	// For Loading in game art as the user clicks on the system. 
+	// Loads one per frame, or if specified to load on frame x.
+	if (mLoadFrame >= mLoadFrameKey) {
+		mGrid.dynamicImageLoader();
+		mLoadFrame = 0;
+	}
+
+	mDescContainer.update(deltatime);
+	mLoadFrame++;
+
+	mGrid.update(deltatime);
+
+	if (mSystemQuickChangeCoolDown < 30)
```

Comment:
I removed a part of this feature. In the function to handle inputs I removed that code to move it up in the mother class ISimpleGameListView: 
```

	// Quick system change
	if (config->isMappedTo("LeftShoulder", input) || config->isMappedTo("LeftTrigger", input) ||
		config->isMappedTo("LeftTop", input) || config->isMappedTo("LeftBottom", input)) {
		if (Settings::getInstance()->getBool("QuickSystemSelect") && mSystemQuickChangeCoolDown > 10) {
			onFocusLost();
			ViewController::get()->goToPrevGameList();
			return true;
		}
	}
```

it allow the user to always switch systems with LeftShoulder and RightShould whatever game view he is in (basic grid or detailed), that way we don't break his habits if he change the type of game view he use.

I will remove the others references to that var "mSystemQuickChangeCoolDown" since it is unused now.